scripts/hammer_command_tree.py

```diff
@@ -6,14 +6,14 @@
 
 from robottelo import ssh
 from robottelo.cli import hammer
+from robottelo.config import settings
 
 
 def generate_command_tree(command):
     """Recursively walk trhough the hammer commands and subcommands and fetch
     their help. Return a dictionary with the contents.
-
     """
-    output = ssh.command(f'{command} --help').stdout
+    output = ssh.command(f'{command} --help', hostname=settings.server.hostnames[0]).stdout
```

Comment:
This should be `settings.server.hostname` but you need to explicitly pass this, the `ssh.command` is smart enough to fetch it if you dont provide the hostname.

Reply:
Hi @jyejare , reading " but you **need** to explicitly pass this" and then " smart enough to fetch it if you dont provide" makes me wonder if you meant " but you **don't need** to explicitly pass this"?

Reply:
@jyejare we set `settings.server.hostname` in [pytest_fixtures/xdist.py](https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/xdist.py#L32), which is in the scope of pytest, and this is just a Python script which is to run locally.
So for `ssh.command` here we need `hostname` param to point to the satellite to check available commands.

and yes, git does not render large diffs by default, but you could check raw file to see that json file https://raw.githubusercontent.com/SatelliteQE/robottelo/a50a99398e81e1cd6d3e421ee700710ba74f879f/tests/foreman/data/hammer_commands.json

Reply:
@swadeley I mean to say he need *not* to pass hostname explicitely !

Reply:
@Gauravtalreja1 Ohh, if this script doesn't run under the pytest session scope, are you passing the satellite server hostname in `conf/server.yaml` manually ?

Reply:
@jyejare Yes, you're correct, we run this script once per y-stream to generate a new hammer_commands.json, which will have updated commands/options in it.